commit c06015148fa9a3cc452ec7121b8c3f59f4a7d6ac
Author: Paul Brook <paul@codesourcery.com>
Date:   Sun Sep 24 16:54:40 2006 +0100

    [ARM] 3860/1: Versatile PCI config byte accesses
    
    The ARM Versatile board PCI config space read routines are broken for byte
    accesses.  The access uses a byte read, so masking the bottom two bits of the
    address is wrong.
    
    I guess this is a cut/paste error from the the halfword code which uses
    aligned word access+shift+mask.
    
    Signed-off-by: Paul Brook <paul@codesourcery.com>
    Signed-off-by: Russell King <rmk+kernel@arm.linux.org.uk>

--- linux-2.6-2.6.18/arch/arm/mach-versatile/pci.c.orig	2006-11-15 01:33:12.657102390 +0100
+++ linux-2.6-2.6.18/arch/arm/mach-versatile/pci.c	2006-11-15 01:33:19.435799210 +0100
@@ -117,7 +117,6 @@
 	} else {
 		switch (size) {
 		case 1:
-			addr &= ~3;
 			v = __raw_readb(addr);
 			break;
 
